tests/Resources/ti.geolocation.test.js

```diff
@@ -307,12 +304,135 @@ describe.windowsBroken('Titanium.Geolocation', () => {
 			it('is a Function', () => {
 				should(Ti.Geolocation).have.a.property('getCurrentHeading').which.is.a.Function();
 			});
+
+			it('works via callback argument', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				// TODO: Do we need location permissions on Android for heading data?
+				Ti.Geolocation.requestLocationPermissions(Ti.Geolocation.AUTHORIZATION_ALWAYS, function (e) {
+					if (!e.success) {
+						return finish(new Error('Failed to get Geolocation ALWAYS permission'));
+					}
+
+					Ti.Geolocation.getCurrentHeading(function (data) {
+						try {
+							if (Ti.Geolocation.hasCompass) {
+								should(data).have.property('success').which.is.true();
+								should(data).have.property('code').which.eql(0);
+								should(data.heading).be.an.Object();
+								should(data.heading).have.a.property('magneticHeading').which.is.a.Number();
+								should(data.heading).have.a.property('accuracy').which.is.a.Number();
+							} else {
+								should(data).have.property('success').which.is.false();
+								should(data).have.property('code').which.eql(1);
+								should(data.error).be.a.String(); // Do we want to verify what it says?
+							}
+						} catch (err) {
+							return finish(err);
+						}
+						finish();
+					});
+				});
+			});
+
+			it('works via Promise return value', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				Ti.Geolocation.requestLocationPermissions(Ti.Geolocation.AUTHORIZATION_ALWAYS, function (e) {
+					if (!e.success) {
+						return finish(new Error('Failed to get Geolocation ALWAYS permission'));
+					}
+					const result = Ti.Geolocation.getCurrentHeading();
+					result.should.be.a.Promise();
+					if (Ti.Geolocation.hasCompass) {
+						result.then(data => {
+							should(data).have.property('success').which.is.true();
+							should(data).have.property('code').which.eql(0);
+							should(data.heading).be.an.Object();
+							should(data.heading).have.a.property('magneticHeading').which.is.a.Number();
+							should(data.heading).have.a.property('accuracy').which.is.a.Number();
+							return finish();
+						}).catch(e => finish(e));
+					} else {
+						// expect to fail if no compass!
+						result.then(() => finish(new Error('Expected to fail getCurrentHeading() with no compass support!'))).catch(e => finish());
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.geolocation.test.js;
  Line: 358;
-->

- :warning: tests/Resources/ti.geolocation.test.js line 358 – 'e' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)